EmployeeManagementSystem/BusinessLogicLayer/Services/DepartmentService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using AutoMapper;
+using BusinessLogicLayer.Interface;
+using BusinessLogicLayer.Models;
+using DataAccessLayer.Interface;
+
+namespace BusinessLogicLayer.Services
+{
+    public class DepartmentService : IDepartmentService
+    {
+        private readonly IDepartmentRepository _repository;
+        private readonly IMapper _mapper;
+        private readonly IEmployeeRepository _employeeRepository;
+        public DepartmentService(IDepartmentRepository repository, IMapper mapper, IEmployeeRepository employeeRepository)
+        {
+            _repository = repository;
+            _mapper = mapper;
+            _employeeRepository = employeeRepository;
+        }
+        public void Create(Department department)
+        {
+            _repository.Create(_mapper.Map<DataAccessLayer.Entities.Department>(department));
+        }
+
+        public void Delete(int id)
+        {
+            var employees = _employeeRepository.GetAll(e => e.DepartmentId == id);
+            foreach (var employee in employees)
+            {
+                employee.DepartmentId = -1;
```

Comment:
why do you need to set DepartmentId -1?

Reply:
It sets to default department with id=-1 and name="No department"

Reply:
okay, if you expect such behavior that you may have employees without a department, then it's better to put this hardcoded ID separately in a constant. but why did you decide not to use null for this?

Reply:
I think it was left from the previous task where I displayed the whole department entity with the employee. So maybe I really don't need this now and will change it to null